internal/controlplane/handlers_oauth.go

```diff
@@ -115,55 +127,93 @@ func (s *Server) GetAuthorizationURL(ctx context.Context,
 	}, nil
 }
 
-// ExchangeCodeForTokenCLI exchanges an OAuth2 code for a token
-// This function gathers the state from the database and compares it to the state
-// passed in. If they match, the code is exchanged for a token.
-// This function is used by the CLI client.
-func (s *Server) ExchangeCodeForTokenCLI(ctx context.Context,
-	in *pb.ExchangeCodeForTokenCLIRequest) (*httpbody.HttpBody, error) {
+// HandleProviderCallback handles the OAuth 2.0 authorization code callback from the enrolled
+// provider. This function gathers the state from the database and compares it to the state
+// passed in. If they match, the provider code is exchanged for a provider token.
+// note: this is an HTTP only (not RPC) handler
+func (s *Server) HandleProviderCallback() runtime.HandlerFunc {
```

Comment:
Do we want to try to enforce that this is HTTPS-only by checking the [`x-forwarded-proto` HTTP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) or [`Forwarded: ...proto`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded)?  (Yeah, that's kinda crummy, but is mostly-standardized, I think)

Reply:
From what I can tell the handlers that are generated from our rpc routes don't enforce HTTPS-only (but maybe I didn't look in the right places), so I don't think we need to enforce it here.